internal/testrunner/runners/system/runner.go

```diff
@@ -948,10 +1001,24 @@ func (r *runner) prepareScenario(ctx context.Context, config *testConfig, svcInf
 	// store the time just before adding the Test Policy, this time will be used to check
 	// the agent logs from that time onwards to avoid possible previous errors present in logs
 	scenario.startTestTime = time.Now()
+	suffixDatastream := svcInfo.Test.RunID
 
 	logger.Debug("adding package data stream to test policy...")
-	ds := createPackageDatastream(*policyToTest, *scenario.pkgManifest, policyTemplate, *scenario.dataStreamManifest, *config, svcInfo.Test.RunID)
-	if r.options.RunTearDown || r.options.RunTestsOnly {
+	policyTesting := kibana.Policy{
+		Name:        fmt.Sprintf("ep-one-test-system-%s-%s-%s", r.options.TestFolder.Package, r.options.TestFolder.DataStream, scenario.startTestTime.Format("20060102T15:04:05Z")),
+		Description: fmt.Sprintf("test policy created by elastic-package test system for data stream %s/%s", r.options.TestFolder.Package, r.options.TestFolder.DataStream),
+		Namespace:   common.CreateTestRunID(),
+	}
+	policyToAssignDatastreamTests := policyToTest
+	if r.shouldCreateNewAgentPolicyForTest() {
```

Comment:
If I just update the `policyToTest` to be like this one, that means that the namespace is random, it would work just while the system tests are not run per stages (--setup, --no-provision, --tear-down).

When these flags are used it would be used the same data stream created in the first run, retrieved from the service state:
https://github.com/elastic/elastic-package/blob/858808bfec7b3dee5de84cdbbbf71ab09b87c09a/internal/testrunner/runners/system/runner.go#L859

That would mean that the method to delete old documents from the data stream could not be delete since that data stream would contain documents for sure.

Not tested, maybe it could be created always the `policyToTest` and just used the one from the state file to assign back the policy to the agent in the scenarios where the tests are performed in stages.

And for those test runs without these flags (common usage), there would be data streams created for each run:
![data streams created](https://github.com/elastic/elastic-package/assets/5330827/e03f7a06-534c-4a4e-9d8e-9c50179000c5)

It could be added the logic to remove those data streams in one of the handlers.

In this PR these 3 policies are used as following:
- one for enrolling without any package policy
    - there were having issues when the stack is running for more than X minutes that agents were not able to enroll using the default policy.
- one to assign the package policy to the agent
    - this one would be used to assign back when running per stages after each --no-provision run 
- one for the actual testing
    - this policy is the one that would be using the Elastic Agent to run the system tests.

Reply:
I'll give it a try to remove one agent policy. I'll need to test with system test running per stages too.